src/Microsoft.Extensions.Caching.Memory/CacheEntryHelper.cs

```diff
@@ -51,9 +51,12 @@ internal static CacheEntry Current
             {
                 if (Scopes != null)
                 {
-                    if (Scopes.Count > 0)
```

Comment:
You shouldn't need any locks in this class, these should all be thread/async local.